packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -95,60 +126,181 @@ function queueImageResizing({ file, args = {}, reporter }) {
   const prefixedSrc =
     options.pathPrefix + `/static/${digestDirPrefix}` + encodedImgSrc
 
-  // Create job and add it to the queue, the queue will be processed inside gatsby-node.js
-  const job = {
-    args: options,
-    inputPath: file.absolutePath,
-    contentDigest: file.internal.contentDigest,
-    outputDir,
-    outputPath: outputFilePath,
+  return {
+    src: prefixedSrc,
+    outputDir: outputDir,
+    relativePath: outputFilePath,
+    width,
+    height,
+    aspectRatio,
+    options,
   }
+}
 
-  const outputFile = path.join(job.outputDir, job.outputPath)
-  let finishedPromise = Promise.resolve()
+async function createJob(job, { reporter }) {
+  const progressBar = createOrGetProgressBar(reporter)
 
-  if (cachedOutputFiles.has(outputFile)) {
-    finishedPromise = cachedOutputFiles.get(outputFile)
+  if (pendingImagesCounter === 0) {
+    progressBar.start()
   }
 
-  // Check if the output file already exists or already is being created.
-  // TODO: Remove this when jobs api is stable, it will have a better check
-  if (!fs.existsSync(outputFile) && !cachedOutputFiles.has(outputFile)) {
-    // schedule job immediately - this will be changed when image processing on demand is implemented
-    finishedPromise = scheduleJob(
-      job,
-      boundActionCreators,
-      pluginOptions,
-      reporter
-    ).then(res => {
-      cachedOutputFiles.delete(outputFile)
+  const transforms = job.args.operations
+  pendingImagesCounter += transforms.length
+  progressBar.total = pendingImagesCounter
 
-      return res
-    })
-
-    cachedOutputFiles.set(outputFile, finishedPromise)
+  if (boundActionCreators.createJobV2) {
+    await boundActionCreators.createJobV2(job)
+  } else {
+    await scheduleJob(job, boundActionCreators)
   }
 
+  progressBar.tick(transforms.length)
+}
+
+function queueImageResizing({ file, args = {}, reporter }) {
```

Comment:
NIT: with `queueImageResizing` not being called anywhere else in this anymore (other than via exported `resize` and `queueImageResizing` functions), could we rename this to `resize` for clarity?

Reply:
Technically we're scheduling a resize 😬. So unsure about this.

Reply:
Hah, if this seems controversial then let's leave as-is ;)